pep-0707.rst

```diff
@@ -0,0 +1,389 @@
+            }
+        }
+        else if (PyCFunction_Check(exit_func)) {
+            if (PyCFunction_GET_FLAGS(exit_func) == METH_O) {
+                /* C function declared as single-arg */
+                return false;
+             }
+        }
+        return true;
+    }
+
+It is important to note that this is not a generic introspection function, but
+rather one which is specifically designed for our use case. We know that
+``exit_func`` is an attribute of the context manager class (taken from the
+type of the object that provided ``__enter__``), so it is typically an unbound method.
+Furthermore, for this to be useful we need to identify enough single-arg forms,
+but not necessarily all of them.  What is critical for backwards compatibility is
+that we will never misidentify a legacy ``exit_func`` as a single-arg one. So,
+for example, ``__exit__(self, *args)`` and ``__exit__(self, exc_type, *args)``
+both have the legacy form, even though they *could* be invoked with one arg.
+
+In summary, an ``exit_func`` will be invoke with a single arg if:
+
+* It is a ``PyMethod`` with ``argcount`` ``2`` (to count ``self``) and no vararg, or
+* it is a ``PyCFunction`` with the ``METH_O`` flag.
+
+Note that any performance cost of the introspection can be mitigated via
+:pep:`specialization <564>`, so it won't be a problem if we need to make it more
+sophisticated than this for some reason.
+
+
+Backwards Compatibility
+=======================
+
+All context managers that previously worked will continue to work in the
+same way because the interpreter will call them with three args whenever
+they can accept three args. There may be context managers that previously
+did not work because their ``exit_func`` expected one argument, so the call
+to ``__exit__`` would have caused a ``TypeError`` exception to be raised,
+and now the call would succeed. This could theoretically change the
+behaviour of existing code, but it is unlikely to be a problem in practice.
+
+The backwards compatibility concerns will show up in some cases when libraries
+try to migrate their context managers from the multi-arg to the single-arg
+signature. If ``__exit__`` or ``__aexit__`` is called by any code other than
+the interpreter's eval loop, the introspection does not automatically happen.
+For example, this will occur where a context manager is subclassed and its
+``__exit__`` method is called directly from the derived ``__exit__``. Such
+context managers will need to migrate to the single-arg version with their
+users, and may choose to offer a parallel API rather than breaking the
+existing one. Alternatively, a superclass can stay with the signature
+``__exit__(self, *args)``, and support both one and three args. Since
+most context managers do not use the value of the arguments to ``__exit__``,
+and simply allow the exception to propagate onward, this is likely to be the
+common approach.
+
+
+Security Implications
+=====================
+
+I am not awware of any.
```

Comment:
```suggestion
I am not aware of any.
```

I haven't signed the CLA, so you may have to make this edit locally to avoid adding me as a co-author and causing the CLA bot to freak out.